src/Injection/InjectionFactory.php

```diff
@@ -81,6 +82,20 @@ public function newInstance(
         foreach ($resolve->setters as $method => $value) {
             $object->$method($value);
         }
+
+        /** @var MutationInterface $mutation */
+        foreach ($resolve->mutations as $mutation) {
+            if ($mutation instanceof LazyInterface) {
+                $mutation = $mutation();
+            }
+
+            if ($mutation instanceof MutationInterface === false) {
+                throw Exception::mutationDoesNotImplementInterface($mutation);
+            }
+
+            $object = $mutation($object);
```

Comment:
Do you think we need to make sure the class is same for the mutated object here ?